File: project/hyeok-onboarding/src/main/java/com/INNER_CIRCLE_ICD4/innerCircle/survey/request/domain/Survey.java
package com.INNER_CIRCLE_ICD4.innerCircle.survey.request.domain;

import jakarta.persistence.*;
import java.util.ArrayList;
import java.util.List;

@Entity
public class Survey {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String title;

    private String description;

    @OneToMany(mappedBy = "survey", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<Question> questions = new ArrayList<>();

    @OneToMany(mappedBy = "survey", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<Response> responses = new ArrayList<>();

    // === 연관 관계 메서드 ===
    public void addQuestion(Question question) {
        questions.add(question);
        question.setSurvey(this);
    }

    public void addResponse(Response response) {
        responses.add(response);
        response.setSurvey(this);
    }

    // === Getters/Setters ===
    public Long getId() {
        return id;
    }

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public List<Question> getQuestions() {
        return questions;
    }

    public List<Response> getResponses() {
        return responses;
    }
}

File: project/jihochoi-onboarding/QueryBox/src/main/kotlin/com/wlghsp/querybox/domain/survey/Survey.kt
package com.wlghsp.querybox.domain.survey

import com.wlghsp.querybox.ui.dto.SurveyUpdateRequest
import jakarta.persistence.Entity
import support.domain.BaseRootEntity

@Entity
class Survey (
    var title: String, // 설문 조사 이름
    var description: String, // 설문 조사 설명

    val questions: Questions = Questions.of(emptyList()), // 설문 받을 항목
    id: Long = 0L,
) : BaseRootEntity<Survey>(id) {

    fun update(request: SurveyUpdateRequest) {
        this.title = request.title
        this.description = request.description
        this.questions.update(request.questions)
    }

    fun getQuestions(): List<Question> {
        return questions.values()
    }

    companion object {
        fun of(title: String, description: String, questions: Questions): Survey {
            require(title.isNotBlank()) { "설문 제목은 비어 있을 수 없습니다."}

            return Survey(
                title = title,
                description = description,
                questions = questions
            )
        }
    }
}